### PR TITLE
Add a minimal example of using astcenc library API

### PR DESCRIPTION
**Pull request by solidpixel**

### CODE REVIEW

File: Utils/Example/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+cmake_minimum_required(VERSION 3.15)
+include(ExternalProject)
+
+project(astcencoder_example VERSION 1.0.0)
+
+# Command line configuration
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
```
Comment (wasimabbas-arm) [inline Utils/Example/CMakeLists.txt:28]: This is not necessary.

File: Utils/Example/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+# Add the external project and pull out the project directories we need
+
+# Add e.g. "CMAKE_CACHE_ARGS -DISA_SSE41:String=ON" to ExternalProject_Add to
+# set configure settings usually found on the astcenc configure command line.
+# Remember that changing ISA changes the name of the library binary produced.
+
+ExternalProject_Add(astcencoder
```
Comment (wasimabbas-arm) [inline Utils/Example/CMakeLists.txt:41]: I like `ExternalProject` command but I haven't seen much use of this in cmake projects. Usually an `add_subdirectory` is the most common way of using TPIP. With external project every time you want to do a clean rebuild, you nuke the whole download and start again. With subdirectory, the dependency needs to be a "sub-directory" for that to work but all of the following explicit methods like `target_include_directories/target_link_directories` goes away.

File: Utils/Example/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        ${SOURCE_DIR}/Source)
+
+# ... with astcencoder Binary dir on the library path and as a library dep
+target_link_directories(astcenc_example
+    PRIVATE
+        ${BINARY_DIR}/Source)
+
+target_link_libraries(astcenc_example
+    PRIVATE
+        astcenc-native-static)
```
Comment (wasimabbas-arm) [inline Utils/Example/CMakeLists.txt:71]: This is the tricky bit. Any advice on how this target name should be generated. I was thinking perhaps the build system can set a variable in parent scope to the name its trying to build and clients can pick it up.

File: Utils/Example/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#  ----------------------------------------------------------------------------
+
+# CMake configuration
+cmake_minimum_required(VERSION 3.15)
+include(ExternalProject)
+
+project(astcencoder_example VERSION 1.0.0)
```
Comment (wasimabbas-arm) [inline Utils/Example/CMakeLists.txt:22]: Some advice on which options to turn ON/OFF would be nice. Specially `-DCLI=OFF` in this case.

File: Utils/Example/astc_api_example.cpp [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+	// ------------------------------------------------------------------------
+	// Compress the image
+	astcenc_image image;
+	image.dim_x = image_x;
+	image.dim_y = image_y;
+	image.dim_z = 1;
+	image.data_type = ASTCENC_TYPE_U8;
+	uint8_t* slices = image_data;
+	image.data = reinterpret_cast<void**>(&slices);
```
Comment (wasimabbas-arm) [inline Utils/Example/astc_api_example.cpp:115]: Don't think slices is needed here. Just cast `image_data` directly?

File: Utils/Example/CMakeLists.txt (hunk 4 above)
Comment (abbaswasim) [reply Utils/Example/CMakeLists.txt:22]: This is resolved now.

File: Utils/Example/CMakeLists.txt (hunk 2 above)
Comment (solidpixel) [reply Utils/Example/CMakeLists.txt:41]: Might add that as a separate example - I can see both being useful to have.

File: Utils/Example/astc_api_example.cpp (hunk 5 above)
Comment (solidpixel) [reply Utils/Example/astc_api_example.cpp:115]: It expects an array of pointers to slices, so the extra indirection is needed.

**Pull request merged**